flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointMetadataOutputStream.java

```diff
@@ -162,4 +161,18 @@ public FsCompletedCheckpointStorageLocation closeAndFinalizeCheckpoint() throws
             }
         }
     }
+
+    static MetadataOutputStreamWrapper getOutputStreamWrapper(
+            final FileSystem fileSystem, final Path metadataFilePath) throws IOException {
+        try {
+            RecoverableWriter recoverableWriter = fileSystem.createRecoverableWriter();
+            if (fileSystem.exists(metadataFilePath)) {
+                throw new IOException("Target file " + metadataFilePath + " is already exists.");
+            }
+            return new RecoverableStreamWrapper(recoverableWriter.open(metadataFilePath));
+        } catch (Throwable throwable) {
+            LOG.warn("Errors on creating recoverable writer.", throwable);
```

Comment:
Perhaps not output the stack here to avoid cause confusion. 

Might change to 

`LOG.warn("Errors on creating recoverable writer due to {}, will use the ordinary writer.", throwable.getMessage());`

Reply:
Make sense.